pkg/cmd/datastore.go

```diff
@@ -81,3 +87,56 @@ func NewGCDatastoreCommand(programName string, cfg *datastore.Config) *cobra.Com
 		}),
 	}
 }
+
+func NewRepairDatastoreCommand(programName string, cfg *datastore.Config) *cobra.Command {
+	return &cobra.Command{
+		Use:     "repair",
+		Short:   "executes datastore repair",
+		Long:    "Executes a repair operation for the datastore",
+		PreRunE: server.DefaultPreRunE(programName),
+		RunE: termination.PublishError(func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			// Disable background GC and hedging.
+			cfg.GCInterval = -1 * time.Hour
+			cfg.RequestHedgingEnabled = false
+
+			ds, err := datastore.NewDatastore(ctx, cfg.ToOption())
+			if err != nil {
+				return fmt.Errorf("failed to create datastore: %w", err)
+			}
+
+			for {
+				wds, ok := ds.(dspkg.UnwrappableDatastore)
+				if !ok {
+					break
+				}
+				ds = wds.Unwrap()
+			}
+
+			repairable, ok := ds.(dspkg.RepairableDatastore)
+			if !ok {
+				return fmt.Errorf("datastore of type %T does not support the repair operation", ds)
+			}
+
+			if len(args) == 0 {
+				fmt.Println()
+				fmt.Println("Available repair operations:")
+				for _, op := range repairable.RepairOperations() {
+					fmt.Printf("\t%s: %s\n", op.Name, op.Description)
+				}
+				return nil
+			}
+
+			operationName := args[0]
+
+			log.Ctx(ctx).Info().Msg("Running repair...")
+			err = repairable.Repair(ctx, operationName, true)
```

Comment:
the only use in the codebase has progress bar enabled. I'd suggest remove this flag and always show the progress bar, and if folks complain, we can add a flag to the command, and then instead of passing it as an argument, we use it to instantiate a different implementation

Reply:
Its used in tests to not show things